test/unittests/capi_test.cpp

```diff
@@ -330,7 +330,9 @@ TEST(capi, find_exported_function)
     EXPECT_EQ(function.type.output, FizzyValueTypeI32);
     EXPECT_NE(function.context, nullptr);
     ASSERT_NE(function.function, nullptr);
-    EXPECT_THAT(function.function(function.context, instance, nullptr, 0), CResult(42_u32));
+    fizzy::ExecutionContext ctx;
+    auto* const c_ctx = reinterpret_cast<FizzyExecutionContext*>(&ctx);
```

Comment:
Hmm, is this the intended C API? :)

Reply:
Calling host function wrapper is not part of C API.

Reply:
Well `function.function` is accessible in C, it is the C struct, isn't it ?

What is the goal of this test?

Reply:
Yeah good point, it was possible to call `FizzyExternalFunction` before.

Maybe wrapped external function in `FizzyExternalFunction wrap(fizzy::ExternalFunction external_func)` should call `unwrap(ctx)` only if context is non-null, otherwise create default one.

Reply:
I'm fine with either but should be documented.

Reply:
Can you also leave a comment here in the test too?

Reply:
> There should not be two way of executing a wasm function. You should pick if you want to execute returned pointer to external function. Or having `execute(index)`.

Not sure why it's a big problem, I think it might be convenient and logical: you're calling `fizzy_find_exported_function` and getting a "function" that you can call.

> Anyway, the solution in #771.

So is #771 in your view introducing a problem, that we should avoid?

Reply:
> > There should not be two way of executing a wasm function. You should pick if you want to execute returned pointer to external function. Or having `execute(index)`.
> 
> Not sure why it's a big problem, I think it might be convenient and logical: you're calling `fizzy_find_exported_function` and getting a "function" that you can call.

I agree here, but then there should not be `fizzy_execute()`. I.e. you can only execute exported functions.
In practice, I prefer the default ExecuteContext creation to be done in single place than in multiple places.

> > Anyway, the solution in #771.
> 
> So is #771 in your view introducing a problem, that we should avoid?

No, it looks good. But there are some details there we need to discuss later.

Reply:
So this breaks one of the ways functions are called, #771 will restore it back, but you think in the future on of the ways should be removed?